extensions/mssql/src/sparkFeature/dialog/sparkJobSubmission/sparkJobSubmissionService.ts

```diff
@@ -0,0 +1,187 @@
+                }
+            }
+
+            // Set other files
+            if (submissionArgs.otherFileList && submissionArgs.otherFileList.trim()) {
+                let otherList = submissionArgs.otherFileList.split(';');
+                if (otherList.length > 0) {
+                    options.body['files'] = otherList;
+                }
+            }
+
+            const response = await this._requestPromise(options);
+            if (response && utils.isValidNumber(response.id)) {
+                return response.id;
+            }
+
+            return Promise.reject(new Error(localize('sparkJobSubmission_LivyNoBatchIdReturned',
+            'No Spark job batch id is returned from response.{0}[Error] {1}', os.EOL, JSON.stringify(response))));
+        } catch (error) {
+            return Promise.reject(error);
+        }
+    }
+
+    public async getYarnAppId(submissionArgs: SparkJobSubmissionInput, livyBatchId: string): Promise<LivyLogResponse> {
+        try {
+            let livyUrl = `https://${submissionArgs.host}:${submissionArgs.port}${submissionArgs.livyPath}/${livyBatchId}/log`;
+            let options = {
+                uri: livyUrl,
+                method: 'GET',
+                json: true,
+                rejectUnauthorized: false,
+                // authentication headers
+                headers: {
+                    'Authorization': 'Basic ' + new Buffer(submissionArgs.userName + ':' + submissionArgs.passWord).toString('base64')
+                }
+            };
+
+            const response = await this._requestPromise(options);
+            if (response && response.log) {
+                return this.extractYarnAppIdFromLog(response.log);
+            }
+
+            return Promise.reject(localize('sparkJobSubmission_LivyNoLogReturned',
+            'No log is returned within response.{0}[Error] {1}', os.EOL, JSON.stringify(response)));
+        } catch (error) {
+            return Promise.reject(error);
+        }
+    }
+
+
+    private extractYarnAppIdFromLog(log: any): LivyLogResponse {
+        let logForPrint = log;
+        if (Array.isArray(log)) {
+            logForPrint = log.join(os.EOL);
+        }
+
+        // eg: '18/08/23 11:02:50 INFO yarn.Client: Application report for application_1532646201938_0182 (state: ACCEPTED)'
+        for (let entry of log) {
+            if (entry.indexOf('Application report for') >= 0 && entry.indexOf('(state: ACCEPTED)') >= 0) {
+                let tokens = entry.split(' ');
+                for (let token of tokens) {
+                    if (token.startsWith('application_')) {
+                        return new LivyLogResponse(logForPrint, token);
+                    }
+                }
+            }
+        }
+
+        return new LivyLogResponse(logForPrint, '');
+    }
+}
+
+export class SparkJobSubmissionInput {
+    public setSparkClusterInfo(hadoopConnection: SqlClusterConnection): void {
```

Comment:
Again let's name to the consistent `sqlClusterConnection`.

Reply:
fixed.